certbot/crypto_util.py

```diff
@@ -242,6 +244,100 @@ def valid_privkey(privkey):
+                Details: {1}".format(renewable_cert.cert, e)
+        logger.exception(error_str)
+        raise e
+
+def verify_cert_matches_priv_key(renewable_cert):
+    """Do the private key and cert match?
+
+    :param typ: `storage.RenewableCert`
+
+    :raises OpenSSL.SSL.Error if they don't match
+    """
+    try:
+        with open(renewable_cert.cert) as cert:
+            with open(renewable_cert.privkey) as privkey:
+                privkey = OpenSSL.crypto.load_privatekey(
+                        OpenSSL.crypto.FILETYPE_PEM, privkey.read())
+                cert = OpenSSL.crypto.load_certificate(
+                        OpenSSL.crypto.FILETYPE_PEM, cert.read())
+                context = OpenSSL.SSL.Context(OpenSSL.SSL.SSLv23_METHOD)
+                context.use_privatekey(privkey)
+                context.use_certificate(cert)
+                context.check_privatekey()
+    except OpenSSL.SSL.Error as e:
+        error_str = "verifying the cert located at {0} matches the \
+                private key located at {1} has failed. \
+                Details: {2}".format(renewable_cert.cert,
+                        renewable_cert.privkey, e)
+        logger.exception(error_str)
+        raise e
+
+
+def verify_fullchain(renewable_cert):
+    """Check that fullchain is indeed cert concatenated with chain
+
+    :param typ: `storage.RenewableCert`
+
+    :raises errors.Error if they don't match
+    """
+    try:
+        with open(renewable_cert.chain) as chain:
+            with open(renewable_cert.cert) as cert:
+                with open(renewable_cert.fullchain) as fullchain:
+                    assert (cert.read() + chain.read()) == fullchain.read()
+    except IOError as e:
+        error_str = "reading one of cert, chain, or fullchain has failed: {0}".format(e)
+        logger.exception(error_str)
+        raise e
+    except AssertionError as e:
+        error_str = "fullchain does not match cert + chain for {0}!"
+        error_str = error_str.format(renewable_cert.lineagename)
+        logger.exception(e)
+        raise e
+
+
+def verify_renewable_cert(renewable_cert):
+    """For checking that your certs were not corrupted on disk.
+    Several things are checked:
+        1. signature verification for the cert
+        2. that fullchain matches cert and pem when concatenated
+        3. check that the private key matches the certificate
+
+    :param typ: `storage.RenewableCert`
+
+    :raises errors.Error is verification fails
+    """
+    verification_errors = []
+    possible_exceptions = (IOError, ValueError, AssertionError,
+            OpenSSL.crypto.Error, OpenSSL.SSL.Error, errors.Error)
+    try:
+        verify_renewable_cert_sig(renewable_cert)
+        verify_fullchain(renewable_cert)
+        verify_cert_matches_priv_key(renewable_cert)
+        return
+    except possible_exceptions as error:
+        verification_errors.append(error)
```

Comment:
Since it's impossible for this list to ever have more than one exception in it, why not just raise the exception containing the encountered error in the body of the `except` block? If you do this, you can remove the `verification_errors` list entirely and the `return` statement from the body of the `try` block.